system/jlib/jargv.cpp

```diff
@@ -218,3 +218,41 @@ bool ArgvIterator::matchPathFlag(StringBuffer & option, const char * name)
 }
 
 
+bool ArgvIterator::matchOptionText(StringBuffer & option, const char * name, bool isBoolFlag, bool isShortOption)
+{
+    const char * arg = query();
+    size_t len = strlen(name);
+    if (memcmp(arg, name, len) != 0)
+        return false;
+
+    switch (arg[len])
+    {
+    case '+':
+    case '-':
+    case '=':
+        option.append(arg);
+        return true;
+    case '\0':
+        break;
+    default:
+        // -Ix or -I x
+        if (isShortOption)
+        {
+            option.append(arg);
+            return true;
+        }
+        return false;
+    }
+
+    option.append(arg);
```

Comment:
Just a note.  Would matter for some string related functions, but not really here.  A return value of false does not necessarily mean that the StringBuffer was unchanged, and option.isEmpty() is not the same as a return value of false.

Reply:
I have changed the code - and also picked up a couple of bugs while testing.